### PR TITLE
Rdm 7416

### PR DESCRIPTION
**Pull request by Thor-tech-of-metal**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ccd/RestTemplateConfiguration.java [hunk 1]
```diff
@@ -95,7 +95,11 @@ private HttpClient getHttpClient(final int timeout) {
         cm.setMaxTotal(maxTotalHttpClient);
         cm.closeIdleConnections(maxSecondsIdleConnection, TimeUnit.SECONDS);
         cm.setDefaultMaxPerRoute(maxClientPerRoute);
-        cm.setValidateAfterInactivity(validateAfterInactivity);
+
+        // Temporally 30 value until we can tune the real value then It will be set in the ENV variables.
```
Comment (Thor-tech-of-metal) [inline src/main/java/uk/gov/hmcts/ccd/RestTemplateConfiguration.java]: cm.setValidateAfterInactivity(30);   is the value from validateAfterInactivity attribute and the value defined in the values.yaml as HTTP_CLIENT_VALIDATE_AFTER_INACTIVITY. 

HTTP_CLIENT_VALIDATE_AFTER_INACTIVITY: '0'  --->  Defines period of inactivity in milliseconds after which persistent connections must
                                                  * be re-validated prior to being {@link #leaseConnection(java.util.concurrent.Future,
                                                   *   long, java.util.concurrent.TimeUnit) leased} to the consumer. Non-positive value passed
                                                   * to this method disables connection validation. This check helps detect connections
                                                   * that have become stale (half-closed) while kept inactive in the pool.

File: src/main/java/uk/gov/hmcts/ccd/RestTemplateConfiguration.java [hunk 2]
```diff
@@ -95,7 +95,11 @@ private HttpClient getHttpClient(final int timeout) {
...
+        cm.setValidateAfterInactivity(10);
```
Comment (luigibk) [inline src/main/java/uk/gov/hmcts/ccd/RestTemplateConfiguration.java]: This value is in ms. 10 seems too short

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -110,9 +110,12 @@ management.endpoint.metrics.cache.time-to-live=1000ms
...
 logging.level.uk.gov.hmcts.ccd.domain.service=${DATA_STORE_DEFAULT_LOG_LEVEL:INFO}
 logging.level.uk.gov.hmcts.ccd=${DATA_STORE_DEFAULT_LOG_LEVEL:INFO}
-# logging.level.org.springframework.web=DEBUG
-#logging.level.org.hibernate=INFO
-#logging.level.org.hibernate.type.descriptor.sql=trace
+
+logging.level.org.springframework.web=DEBUG
+logging.level.org.hibernate=INFO
+logging.level.org.hibernate.type.descriptor.sql=trace
+logging.level.org.apache.http=DEBUG
```
Comment (luigibk) [inline src/main/resources/application.properties]: Isn't this too much logging? Is it just for testing? Also how is the hibernate logging level related to this change?

File: src/main/java/uk/gov/hmcts/ccd/RestTemplateConfiguration.java (hunk 2 above)
Comment (Thor-tech-of-metal) [reply src/main/java/uk/gov/hmcts/ccd/RestTemplateConfiguration.java]: The default is 15 . I think that we can try 2

Comment (luigibk) [reply src/main/java/uk/gov/hmcts/ccd/RestTemplateConfiguration.java]: Default is 2000ms from what I can see. Yes, it seems reasonable

File: src/main/java/uk/gov/hmcts/ccd/RestTemplateConfiguration.java [hunk 4]
```diff
@@ -95,7 +95,11 @@ private HttpClient getHttpClient(final int timeout) {
         cm.setMaxTotal(maxTotalHttpClient);
         cm.closeIdleConnections(maxSecondsIdleConnection, TimeUnit.SECONDS);
         cm.setDefaultMaxPerRoute(maxClientPerRoute);
-        cm.setValidateAfterInactivity(validateAfterInactivity);
+
+        // Temporally 2 value until we can tune the real value then It will be set in the ENV variables.
+        cm.setValidateAfterInactivity(2);
```
Comment (luigibk) [inline src/main/java/uk/gov/hmcts/ccd/RestTemplateConfiguration.java]: This should 2000. This value is in milliseconds

**Pull request closed without merging**